Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1428,6 +1428,15 @@ void FrameLoader::loadFrameRequest(FrameLoadRequest&& request, Event* event, Ref
     else
         loadType = FrameLoadType::Standard;
 
+    if (loadType == FrameLoadType::Reload) {
+        if (m_frame->document() && m_frame->document()->settings().navigationAPIEnabled()) {
+            if (RefPtr domWindow = m_frame->document()->domWindow()) {
+                if (!domWindow->protectedNavigation()->dispatchPushReplaceReloadNavigateEvent(url, NavigationNavigationType::Reload, false))
```

Comment:
Can this cause JS re-entry (JS calling C++ which calls JS on the same stack)?  I think it can, in which case this event should be scheduled on the event loop instead of called synchronously.

Reply:
What kind of scenario are you thinking of? Are you worried about recursion/endless loop? My latest patch protects against detachment in the event handler btw.

Reply:
I added the event loop scheduling, PTAL.